Bugsnag/Configuration/BugsnagConfiguration.m

```diff
@@ -415,6 +417,26 @@ -(void)deletePersistedUserData {
 // MARK: - Properties: Getters and Setters
 // -----------------------------------------------------------------------------
 
+@synthesize maxPersistedEvents = _maxPersistedEvents;
+
+- (NSUInteger)maxPersistedEvents {
+    @synchronized (self) {
+        return _maxPersistedEvents;
+    }
+}
+
+- (void)setMaxPersistedEvents:(NSUInteger)maxPersistedEvents {
+    @synchronized (self) {
+        if (maxPersistedEvents <= 100) {
+            _maxPersistedEvents = maxPersistedEvents;
+        } else {
+            bsg_log_err(@"Invalid configuration value detected. Option maxPersistedEvents "
+                        "should be an integer between 0-100. Supplied value is %lu",
+                        (unsigned long) maxPersistedEvents);
+        }
```

Comment:
1. Can `notify` still function (e.g. actually deliver a report) if `maxPersistentEvents` is set to 0? if not, I'd recommend that for cocoa this value should be between 1-100 rather than 0-100.

2. Perhaps attempting to set an invalid value should follow the intent instead of being ignored? So if I set the value to 500, there will be a warning and it will be set to 100.

Reply:
Unresolving for the latter half:

Perhaps attempting to set an invalid value should follow the intent instead of being ignored? So if I set the value to 500, there will be a warning and it will be set to 100.

Reply:
We'd have to change the rest of the checks to match this pattern then. This code is basically copied from another check in BugsnagConfiguration, which refuses to change the value if it's out of range.

Reply:
Making the change elsewhere may not be a bad thing, and can be done incrementally. For now I'm asking 1) whether it was considered and 2) if it would be the ideal behavior, since this is new functionality and we have the opportunity to get it right without consequence.

Reply:
I'm not sure. We should probably be harmonizing the behaviour across systems. This code is copied from the breadcrumbs system. Specifically:

- Setting to an out-of-range value prints an error and refuses to set the value.
- Setting to 0 effectively disables breadcrumbs.

Is this the behaviour we want?